packages/rulesets/src/asyncapi/functions/utils/mergeTraits.ts

```diff
@@ -0,0 +1,36 @@
+import { isPlainObject } from '@stoplight/json';
+
+type HaveTraits = { traits?: any[] } & Record<string, any>;
+
+export function mergeTraits<T extends HaveTraits>(data: T): T {
```

Comment:
I would probably add some more docs here and specifically stating that this is JSON Merge Patch.

https://www.rfc-editor.org/rfc/rfc7386

Reply:
Done, thanks! :)